src/main/java/graphql/execution/MergedField.java

```diff
@@ -56,30 +59,24 @@
  * }
  * </pre>
  * These examples make clear that you need to consider all merged fields together to have the full picture.
- *
+ * <p>
  * The actual logic when fields can be successfully merged together is implemented in {#graphql.validation.rules.OverlappingFieldsCanBeMerged}
  */
 @PublicApi
+@NullMarked
 public class MergedField {
 
-    private final ImmutableList<Field> fields;
     private final Field singleField;
     private final ImmutableList<DeferredExecution> deferredExecutions;
 
-    private MergedField(ImmutableList<Field> fields, ImmutableList<DeferredExecution> deferredExecutions) {
```

Comment:
We have two classes now - a MergedField and a MultiMergedField and the second is only used when we have multiple fields and hence burns a list but otherwise its a single value